sljit_src/sljitNativePPC_common.c

```diff
@@ -666,12 +666,18 @@ SLJIT_API_FUNC_ATTRIBUTE void* sljit_generate_code(struct sljit_compiler *compil
 
 	code_ptr = (sljit_ins *)SLJIT_ADD_EXEC_OFFSET(code_ptr, executable_offset);
 
-	SLJIT_CACHE_FLUSH(code, code_ptr);
-	SLJIT_UPDATE_WX_FLAGS(code, code_ptr, 1);
-
 #if (defined SLJIT_INDIRECT_CALL && SLJIT_INDIRECT_CALL)
+	compiler->executable_size = (sljit_uw)(code_ptr - code) * sizeof(sljit_ins) + sizeof(struct sljit_function_context) / sizeof(sljit_ins);
+	SLJIT_CACHE_FLUSH(code, code_ptr + sizeof(struct sljit_function_context) / sizeof(sljit_ins));
```

Comment:
Why this change is added? The function context is read only memory structure, and it is never executed. The size computation here is a good idea.